### PR TITLE
Promise request options

### PR DESCRIPTION
**Pull request by South-Paw**
# Why

Second go at allowing for promised request options to be provided to components and hooks.

See #180 for more detail

### CODE REVIEW

File: src/Poll.tsx [hunk 1]
```diff
@@ -343,13 +343,13 @@ function Poll<TData = any, TError = any, TQueryParams = { [key: string]: any }>(
   // Compose Contexts to allow for URL nesting
   return (
     <RestfulReactConsumer>
-      {contextProps => {
+      {async contextProps => {
         const contextRequestOptions =
           typeof contextProps.requestOptions === "function"
-            ? contextProps.requestOptions()
+            ? await contextProps.requestOptions()
```
Comment (South-Paw) [inline src/Poll.tsx]: I saw you didn't add an await to this one in your commits @fabien0102 - was that accidental or did you intend to not await this one?

File: src/Poll.tsx [hunk 2]
```diff
@@ -343,13 +343,13 @@ function Poll<TData = any, TError = any, TQueryParams = { [key: string]: any }>(
   // Compose Contexts to allow for URL nesting
   return (
     <RestfulReactConsumer>
-      {contextProps => {
+      {async contextProps => {
```
Comment (South-Paw) [inline src/Poll.tsx]: The poll component needs some kind of rework to support this I think.

Placing this async here is what creates all the invariant errors we see in the tests - React tries to render the Promise object of this function

I'm unsure how you want to handle this?

**Comment by South-Paw** [comment]
@fabien0102 look! it's all green 😄

second times a charm it seems!

**Comment by fabien0102** [comment]
Published in `restful-react@9.5.0`

**Pull request merged**